### DISCUSSION TITLE
how do I even use this??

### DISCUSSION THREAD

**Discussion by prince60709** [gilzoide/lua-gdextension#44] [Q&A]
I go to the godot asset library within my godot project and download this extension, but how do I actually use it to write Lua script?? do I have to do something with the code put in the readme? 

I am on a Mac, so do I have to give "explicit permission" to something or "codesign" something? if that's the case, what even is codesigning and what kind of permission do I need to give to what??

I'm sorry if this seems like a stupid question to whoever is reading this; I am genuinly perplexed, tho

**Comment by TrainzMarcel** [comment]
sorry about the late response, did u go into the project settings and enable the plugin?

**Reply by gilzoide** [reply]
There are no plugins in this project yet, like Lua PluginScript does.
In Lua PluginScript, it isn't even 100% necessary: all it does is minify Lua code when exporting, as well as providing the REPL tab for trying out Lua code in the editor.

**Comment by gilzoide** [comment]
Hey @Prince60709, sorry about the late response as well.

Lua GDExtension for now simply creates classes for interacting GDScript with Lua code "manually".
There is currently no way to write Godot scripts directly in Lua like [Lua PluginScript](https://github.com/gilzoide/godot-lua-pluginscript) did for Godot 3.X, or like [godot_luaAPI](https://github.com/WeaselGames/godot_luaAPI) or [luascript](https://github.com/perbone/luascript) do.
Lua GDExtension is **_not_** the right tool (yet) for scripting Godot objects in Lua. (I say "yet" because there has been some time invested in registering Lua as a scripting language, but I haven't worked in that for a long time and it doesn't work yet)

For now, the supported use case is: you have a game/app that wants to call Lua code, perhaps code that users create, maybe for modding, maybe part of the gameplay. One could make a game that teaches coding using Lua scripts, for example.

The [Calling Lua from Godot example](https://github.com/gilzoide/lua-gdextension?tab=readme-ov-file#calling-lua-from-godot) demonstrates this "manual" approach, which for now is the only approach available.

> I am on a Mac, so do I have to give "explicit permission" to something or "codesign" something? if that's the case, what even is codesigning and what kind of permission do I need to give to what??

Yeah, so for the "permission" thing you need to right-click the `libluagdextension.macos.template_debug.universal.dylib` file, then click "Open", then a popup will appear telling you that this executable is not trusted by the system and you choose "Open Anyway". If you want more info about this security system, it is called the "Gatekeeper".
Codesigning is a way to tell the mac which developer signed the executable, so that it can be trusted. The mac won't trust an unsigned executable, unless you built it yourself. I can try signing it with a developer signature, but I don't know if another mac that is not mine will trust it (I definitely need to test this, for the sake of all plugins I make).
